### ISSUE TITLE
Error when trying to add new product

### ISSUE THREAD

**Issue by madeincosmos** [Automattic/wp-calypso#30884]
#### Steps to reproduce
1. Starting at URL: Originally reported for yehey.com, reproduced on a new site at mariascosmicshop.wpcomstaging.com
2. Go to **My Site > Store**,
3. Click on **Products > Add**,
4. Observe the page just reloads without letting you add new product,
5. Click on **Products > Add** again,
6. Observe the error:

![Screenshot](https://cld.wthms.co/Ki9pop+)
Screenshot: https://cld.wthms.co/Ki9pop

#### What I expected
To add a new product

#### What happened instead
There's an error in the console 

```
vendors~build.a6fb05f1f75b9d87d809.min.js:71 TypeError: Cannot read property 'state' of undefined
```

#### Browser / OS version
Chrome 72, OS X 10.13.4

#### Screenshot / Video
![Screenshot](https://cld.wthms.co/Ki9pop+)
Screenshot: https://cld.wthms.co/Ki9pop

**Comment by TeniCola** [comment]
Thank you for adding this issue, I had a similar report from a couple users in the past day and was planning to write one in between my core shifts.

HC-10560524 
HC-10564491 

One of users (HC-10564491) I believe had the same issue @madeincosmos reported with the error page, but the other user (10560524) got a blank page instead when they tried to edit or add a new product thru the store:

> All of the sudden this morning, I cannot add or edit product listings. When I click the ADD button [...] it just refreshes the page. When I go to [the Product page] there is nothing, just a blank screen.

I was able to replicate this blank screen in SU-mode during the livechat, but I can't replicate it now while trying to access their site again for a screenshot. The Store feature is currently working as expected for them now, so this bug seems to be one that comes and goes.

At the time of the live chat however, a hard refresh / browser switch did not fix the issue. I asked both users to access their Products thru the wp-admin dashboard, as a temporary solution until we fixed the issue.

Slack discussion with other HEs and Users mentioning this problem: p1550592864129900-slack-livechat
